client/transport/cancellable/cancellable.go

```diff
@@ -0,0 +1,107 @@
+// Copyright 2015 The Go Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+// Package cancellable provides helper function to cancel http requests.
+package cancellable
```

Comment:
Looks like this is `golang.org/x/net/context/ctxhttp`. Why is it being forked instead of vendored?

If there are modifications from upstream, those should be noted.